assets/wizards/advertising/views/providers/index.js

```diff
@@ -78,33 +80,37 @@ const Providers = ( { services, fetchAdvertisingData, toggleService } ) => {
 
 	return (
 		<>
-			<ActionCard
-				title={ __( 'Google Ad Manager', 'newspack-plugin' ) }
-				description={ __(
-					'Manage Google Ad Manager ad units and placements directly from the Newspack dashboard.',
-					'newspack-plugin'
-				) }
-				actionText={
-					google_ad_manager && google_ad_manager.enabled && __( 'Configure', 'newspack-plugin' )
-				}
-				toggle
-				toggleChecked={ google_ad_manager && google_ad_manager.enabled }
-				toggleOnChange={ value => {
-					toggleService( 'google_ad_manager', value ).then( () => {
-						if (
-							value === true &&
-							! google_ad_manager.status.connected &&
-							! google_ad_manager.status.network_code
-						) {
-							setIsOnboarding( true );
-						}
-					} );
-				} }
-				titleLink={ google_ad_manager?.enabled ? '#/google_ad_manager' : null }
-				href={ google_ad_manager?.enabled && '#/google_ad_manager' }
-				notification={ notifications.length ? notifications : null }
-				notificationLevel={ google_ad_manager.created_targeting_keys?.length ? 'success' : 'error' }
-			/>
+			{ newspack_ads_wizard.show_gam_sa_setup && (
```

Comment:
This condition overreaches. If I force a truthy `show_gam_sa_setup` value to simulate a Newspack-hosted site, it ends up hiding the entire GAM card, which makes it impossible to connect a site that's not already connected to GAM.

<img width="1752" alt="Screenshot 2024-03-14 at 11 07 04 AM" src="https://github.com/Automattic/newspack-plugin/assets/2230142/9142ce9c-623a-4ade-95cb-c2a1421ca062">

Instead of hiding the entire card, we should change the behavior of the "Click here to connect your account" link to redirect to the Connections page for Newspack-hosted sites.

Reply:
Thanks for catching this! I thought if the site was connected to manager, we wanted to use the Connections wizard to onboard, but after looking more deeply at the code, I realized we had already accounted for the oauth connection flow in the Advertising wizard, but the problem was the oauth flow was always failing and falling back to the SA upload flow. This was happening because the wizard was always attempting to connect via oauth on render, even when the account hadn't been onboarded yet.

I've made some changes in https://github.com/Automattic/newspack-plugin/pull/2986/commits/8c052b671342db4b80bc6b4398d8a6cf62edd23b to correctly display the correct onboarding step in advertising wizard depending on connection to manager. I've also updated the testing steps to reflect the new changes.